File: frontend/svelte-app/src/lib/utils/assistantModelConfig.js
// Only enabled model IDs from the authoritative capability list can be defaults.
export function reconcileModelDefaults(defaults, capabilities) {
 // Creator resolves the same model choice for form, CLI and AAC.
 if (capabilities?.model_defaults) return { ...defaults, ...capabilities.model_defaults };
 const connectors = capabilities?.connectors || {};
 const models = (name) => connectors[name]?.available_llms || [];
 let connector = defaults.connector;
 if (!models(connector).length) {
  // Never silently select a debug connector for a real assistant.
  connector = Object.keys(connectors).find(name => name !== 'bypass' && models(name).length) || '';
 }
 const llm = models(connector).includes(defaults.llm) ? defaults.llm : (models(connector)[0] || '');
 return { ...defaults, connector, llm };
}

File: backend/lamb/assistant_model_config.py
"""Model selection for new-assistant forms and configuration clients."""

def model_configuration(capabilities, form_defaults, global_default):
    connectors = capabilities.get('connectors', {})
    def models(provider):
        return connectors.get(provider, {}).get('available_llms', [])
    provider, model = global_default.get('provider', ''), global_default.get('model', '')
    available = bool(provider and model and model in models(provider))
    if available:
        selected_provider, selected_model = provider, model
    else:
        selected_provider = form_defaults.get('connector', '')
        if not models(selected_provider):
            selected_provider = next((name for name in connectors if name != 'bypass' and models(name)), '')
        preferred = form_defaults.get('llm', '')
        selected_model = preferred if preferred in models(selected_provider) else next(iter(models(selected_provider)), '')
    return {
        'global_default_model': {'provider': provider, 'model': model},
        'global_default_available': available,
        'model_defaults': {'connector': selected_provider, 'llm': selected_model},
    }
